### PR TITLE
A first pass at making some of the device information configurable from boards.txt

### PR DESCRIPTION
**Pull request by obra**

### CODE REVIEW

File: cores/arduino/USBCore.cpp [hunk 1]
```diff
@@ -59,8 +74,8 @@ usb_desc_config configDesc = {
   .bNumInterfaces = 0,
   .bConfigurationValue = 1,
   .iConfiguration = 0,
-  .bmAttributes = 0b10000000,
-  .bMaxPower = 50
+  .bmAttributes = 0b100000000,
```
Comment (maxgerhardt) [inline cores/arduino/USBCore.cpp:77]: This PRing *back* the value `.bmAttributes = 256` (overflow with respect to `uint8_t`) which above was fixed to be 128 above. Are you sure about this?

Comment (bjc) [reply cores/arduino/USBCore.cpp:77]: I assume this is an oversight, so I'll revert the change on my end. There's no way 256 is valid here, and the most significant bit is required to be set by USB under all conditions.

File: cores/arduino/USBCore.cpp [hunk 2]
```diff
@@ -33,8 +38,8 @@ static usb_desc_dev devDesc = {
   // TODO: this depends on what the mcu can support, but this is
   // device dependent code, so nevermind?
   .bMaxPacketSize0       = USBD_EP0_MAX_SIZE,
-  .idVendor              = USBD_VID,
-  .idProduct             = USBD_PID,
+  .idVendor              = USB_VID,
+  .idProduct             = USB_PID,
```
Comment (bjc) [inline cores/arduino/USBCore.cpp]: I had changes that accomplished this in 0a3dadbb961cfd1a25d13ef8e3830e3812335415

**Review by bjc — COMMENTED** [review]
It looks like the only thing you've added wasn't covered already was the `USB_CONFIG_POWER_MA` &c macros, so I've added those by hand; it seemed easier than trying to do a merge just to pull stuff out afterwards, or asking you to resubmit.

Did I miss anything else?

**Pull request closed without merging**